### PR TITLE
Add wireguard module to kernel

### PR DESCRIPTION
**Pull request by eriknordmark**
This enables OCI contrainers running on EVE-OS to use wireguard.
Those are run using the same kernel as we have in the host.
At some point in time maybe we want a separate kernel for the host and the container runtime.

### CODE REVIEW

**Review by deitch — APPROVED** [review]
LGTM, although CI is not happy.

**Review by giggsoff — CHANGES_REQUESTED** [review]
Please adjust riscv64 kernel config: [0001-Fix-riscv64-kernel-config.txt](https://github.com/lf-edge/eve/files/9061185/0001-Fix-riscv64-kernel-config.txt)

**Review by milan-zededa — APPROVED** [review]

File: pkg/kernel/kernel_config-5.10.x-x86_64 [hunk 1]
```diff
@@ -2186,7 +2186,8 @@ CONFIG_MII=m
 CONFIG_NET_CORE=y
 CONFIG_BONDING=m
 CONFIG_DUMMY=m
-# CONFIG_WIREGUARD is not set
```
Comment (zededa-yuri) [inline pkg/kernel/kernel_config-5.10.x-x86_64:2189]: Out of curiosity. You just did 'make menuconfig' and enabled `CONFIG_WIREGUARD`. And all the other changes are done by linux's configuration utility? Or there is more to that.

The reason I ask, is that I had an urge to introduce config fragments for a little while. Which is a bit hard to do at the moment without knowing the intention of changes which have been done to `.config` in the past.

**Review by zededa-yuri — COMMENTED** [review]
Would you mind adding the

**Review by zedi-pramodh — APPROVED** [review]
LGTM.

**Comment by rouming** [comment]
This seems to me partially copies #2711 . Still the same question: why do we need to load modules from the xen-tools container and not from a rootfs from a single place? It is already quite difficult to figure out the correct booting sequence and module order, and if containers start load modules (which are asynchronous to each other) - this gets really complicated and messy.

**Review by zedi-pramodh — APPROVED** [review]
LGTM

**Pull request merged**